enhancements/20240805-encrypted-block-volume.md

```diff
@@ -0,0 +1,333 @@
+# Support Block Volume Encryption/Decryption
+
+This enhancement focuses on the encryption and decryption of block volumes, particularly involving boot volumes and data volumes. 
+
+
+## Summary
+
+This enhancement introduces encryption and decryption capabilities for block volumes, specifically targeting boot volumes (virtual machine images) and data volumes. This functionality will enhance security by ensuring that data stored in these volumes is encrypted and can be decrypted when attached. 
+
+## Related Issues
+
+https://github.com/harvester/harvester/issues/3129
+
+## Motivation
+
+
+### Goals
+
+- Implement encryption for boot volumes and data volumes (support hot plug). 
+- Provide a clear and detailed setup process for users to enable encryption/decryption. 
+
+### Non-goals
+
+- This enhancement will not cover GUI-related changes. That particularly means we won't provide more convenient way to enable encryption/decryption with GUI in this enhancement. This will be addressed in next enhancement.
+
+
+## Proposal
+
+Since Longhorn uses clone mechanism to encrypt/decrypt Boot volume, so Harvester will follow that. This means that there would be two virtual machine image after encryption. One is encrypted, and the other is source virtual machine image. 
+
+Data volumes are different from Boot volume. Because Longhorn supports block encryption/decryption, so Harvester could encrypt/decrypt data volume directly. 
+
+Regardless of volumes used, Longhorn will decrypt volume when it's attached to virtual machine. So, we'll be able to read the boot volumes and data volumes in the virtual machine. 
+
+### User Stories
+
+#### Story 1 - Encrypt/Decrypt Boot Volumes
```

Comment:
Another story: Users should also be able to upload an encrypted image and create VMs based on it directly.

Reply:
@innobead 

Could you elaborate more about "upload an encrypted image" ?

The original use case aims to prepare a image (upload image or use URL to download), then enable encryption. After that, users could reuse this encrypted image to create VM. I think original title was confused, so I changed it.

Reply:
@innobead No, they can't. We don't support it. If we'd like to do this, we need to add some fields on GUI to let users fill out some encryption related information. Otherwise, backend can't use the longhorn clone mechanism to encrypt image.